src/RaceChrono.cpp

```diff
@@ -39,18 +44,71 @@ void handle_racechrono_filter_request(
 
   // TODO: figure out how to report errors.
 }
+#elif defined(ARDUINO_ARCH_ESP32)
+class handle_racechrono_filter_request: public NimBLECharacteristicCallbacks {
+
+  void onWrite(BLECharacteristic* pCharacteristic) {
+    std::string data = pCharacteristic->getValue();
+    int len = pCharacteristic->getValue().length();
+    if (len < 1) {
+      // TODO: figure out how to report errors.
+      return;
+    }
+
+    switch (data[0]) {
```

Comment:
Instead of duplicating code and having basically the same code in two places, let's extract a helper function, or maybe even a helper class.

Having identical/near-identical code in multiple places is a sure recipe for inconsistencies and device-specific bugs in the future.